javaparser-core/src/main/java/com/github/javaparser/printer/lexicalpreservation/LexicalDifferenceCalculator.java

```diff
@@ -290,9 +290,9 @@ private void calculatedSyntaxModelForNode(CsmElement csm, Node node, List<CsmEle
         } else if ((csm instanceof CsmString) && (node instanceof TextBlockLiteralExpr)) {
             // FIXME: csm should be CsmTextBlock -- See also #2677
             if (change instanceof PropertyChange) {
-                elements.add(new CsmToken(GeneratedJavaParserConstants.TEXT_BLOCK_LITERAL, "\"\"\"" + ((PropertyChange) change).getNewValue() + "\"\"\""));
+                elements.add(new CsmToken(GeneratedJavaParserConstants.TEXT_BLOCK_LITERAL, "\"\"\"\n" + ((PropertyChange) change).getNewValue() + "\"\"\""));
             } else {
```

Comment:
To set the most appropriate eof character, you can probably use Node.getLineEndingStyle().toString().

Reply:
This is not a matter of style, it is a matter of Java specification Just like prettyPrinter adds a `\n`.

On a system where EOL is '\r', `Node.getLineEndingStyle().toString()` would break `TextBlockLiteralExpr`

Reply:
This is similar to `CsmTextBlock.prettyPrint(Node, SourcePrinter)` not relying on `printer.println()`

Reply:
https://docs.oracle.com/en/java/javase/14/docs/specs/text-blocks-jls.html

Reading https://www.vojtechruzicka.com/java-text-blocks/ makes me feel I overlooked part of the spec

```
Line terminators are normalized to the ASCII LF character, as follows:

1. An ASCII CR character followed by an ASCII LF character is translated to an ASCII LF character.
2. An ASCII CR character is translated to an ASCII LF character.
```

I'm less comfortable with dynamic EOL in this case, but you're the boss.

Should I do the same in `CsmTextBlock.prettyPrint(Node, SourcePrinter)  `?

Reply:
// Note that values within TextBlocks ALWAYS have the \n line ending, per https://openjdk.java.net/jeps/378#1--Line-terminators
        
(`CsmTextBlock` also felt '\n' were preferable)